enumerable_methods.rb

```diff
@@ -89,14 +96,16 @@ def my_any?
   end
 
   def my_none?
-    return to_enum unless block_given?
-
     arr = self
     if arr.is_a? Array
       x = true
       i = 0
       while i < arr.length
-        x = false if yield(arr[i])
+        if block_given?
+          x = false if yield(arr[i])
+        else
+          x = false if arr[i]
+        end
         i += 1
       end
       x
```

Comment:
**[REQUIRED]**
* When a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern.
```
Failure/Error: expect(words.my_any?('cat')).to be words.any?('cat')
     
     ArgumentError:
       wrong number of arguments (given 1, expected 0)
```
* When a Regex is passed as an argument returns true only if none of the collection matches the Regex.
```
Failure/Error: expect(words.my_none?(/z/)).to be words.none?(/z/)
     
     ArgumentError:
       wrong number of arguments (given 1, expected 0)
```
* When a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern.
```
Failure/Error: expect(words.my_none?(5)).to be words.none?(5)
     
     ArgumentError:
       wrong number of arguments (given 1, expected 0)
```